### PR TITLE
Support loading olap table with StarOSTablet

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Problem Summary(Required) ：

1. Select the primary replica backend to write data.
2. No longer check quorum when committing and publishing transaction.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 64 / 88 (72.73%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/StarOSTablet.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/catalog/StarOSAgent.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|12|15|80.00%|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|52|65|80.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 72 / 88 (81.82%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|12|15|80.00%|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|52|65|80.00%|
|:large_blue_circle:|com/starrocks/catalog/StarOSTablet.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/StarOSAgent.java|7|7|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java [hunk 1]
```diff
@@ -479,52 +480,67 @@ public void commitTransaction(long transactionId, List<TabletCommitInfo> tabletC
                     continue;
                 }
 
+                boolean useStarOS = partition.isUseStarOS();
+
                 List<MaterializedIndex> allIndices = transactionState.getPartitionLoadedTblIndexes(tableId, partition);
                 int quorumReplicaNum = table.getPartitionInfo().getQuorumNum(partition.getId());
                 for (MaterializedIndex index : allIndices) {
                     for (Tablet tablet : index.getTablets()) {
                         int successReplicaNum = 0;
```
[the remark this answers is not in the document]
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java]: ok

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 74 / 90 (82.22%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|12|15|80.00%|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|53|66|80.30%|
|:large_blue_circle:|com/starrocks/catalog/StarOSTablet.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/StarOSAgent.java|8|8|100.00%|

**Review by gengjun-git — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Pull request merged**